src/config/imageoptionview.swift

```diff
@@ -0,0 +1,66 @@
+import SwiftUI
+import SwiftyJSON
+
+private let modes = [
+  NSLocalizedString("Local", comment: ""),
+  "URL",
+]
+
+private let imageDir = wwwDir.appendingPathComponent("img")
+
+struct ImageOptionView: OptionView {
+  let label: String
+  let overrideLabel: String? = nil
+  @ObservedObject var model: ImageOption
+  @State private var openPanel = NSOpenPanel()
+
+  func selectImage() {
+    mkdirP(imageDir.localPath())
+    // Only consider the first image, but allow multiple deletion.
+    openPanel.allowsMultipleSelection = true
+    openPanel.canChooseDirectories = false
+    openPanel.allowedContentTypes = [.image]
+    openPanel.directoryURL = imageDir
+    openPanel.begin { response in
+      if response == .OK {
+        guard let file = openPanel.urls.first else {
+          return
+        }
+        var fileName = file.lastPathComponent
+        if !imageDir.contains(file) {
+          if !copyFile(file, imageDir.appendingPathComponent(fileName)) {
+            return
```

Comment:
I think we should add an error message here, at least in the log file

Reply:
copyFile already [logs](https://github.com/fcitx-contrib/fcitx5-macos/blob/e14a44ae651c99b258497e17a8998dfb67558495/src/config/util.swift#L78) on fail.
What I want to improve in the future is to provide a convenient way to trigger an AlertToast from nested subview to window's major view. Currently the only place that does the similar thing is https://github.com/fcitx-contrib/fcitx5-macos/blob/e14a44ae651c99b258497e17a8998dfb67558495/src/config/inputmethod.swift#L215-L228 but I'm too weak Swift programmer to make it flexible 🌚